examples/website/plot/plot-layer/surface-layer.ts

```diff
@@ -1,24 +1,42 @@
 import {Layer, picking} from '@deck.gl/core';
-import GL from '@luma.gl/constants';
-import {Model} from '@luma.gl/core';
+import type {Color, DefaultProps, LayerDataSource, LayerProps} from '@deck.gl/core';
+import {GL} from '@luma.gl/constants';
+import {Model} from '@luma.gl/engine';
+import {ScaleLinear} from 'd3-scale';
 
 import surfaceVertex from './surface-vertex.glsl';
 import fragmentShader from './fragment.glsl';
+import {$TODO} from './types';
 
-const DEFAULT_COLOR = [0, 0, 0, 255];
+const DEFAULT_COLOR: Color = [0, 0, 0, 255];
 
-const defaultProps = {
+const defaultProps: DefaultProps<SurfaceLayerProps> = {
   data: [],
   getPosition: () => [0, 0, 0],
   getColor: () => DEFAULT_COLOR,
-  xScale: null,
-  yScale: null,
-  zScale: null,
+  xScale: undefined,
+  yScale: undefined,
+  zScale: undefined,
   uCount: 100,
   vCount: 100,
   lightStrength: 0.1
 };
 
+/** All props supported by SurfaceLayer. */
+export type SurfaceLayerProps<DataT = unknown> = _SurfaceLayerProps<DataT> & LayerProps;
+
+type _SurfaceLayerProps<DataT> = {
+  data: LayerDataSource<DataT>;
+  getPosition?: $TODO;
```

Comment:
On my branch, I've typed some of these already: https://github.com/visgl/deck.gl/blob/felix/surface-layer-v9/examples/website/plot/plot-layer/surface-layer.ts#L29-L30

Reply:
The issue I ran into here was that the types from the branch require `// @ts-ignore` when using the accessors, and I slightly preferred a `$TODO` at the top over `// @ts-ignore` each time they're used.

I'll take another look and see if I can figure out the correct strict types, though.

Reply:
Got this in a state that TS is happy with, without `// @ts-ignore` or `$TODO`. There might be a less verbose way to do it, but I couldn't find other examples, and probably best to just be verbose and strict in the meantime.